src/instructlab/configuration.py

```diff
@@ -786,125 +682,10 @@ class Config(BaseModel):
-                "llama_cpp": {"gpu_layers": -1, "max_ctx_size": 4096, "llm_family": ""},
-                "vllm": {"llm_family": "", "vllm_args": []},
-            },
-            "generate": {
-                "model": DEFAULTS.DEFAULT_MODEL,
-                "taxonomy_path": DEFAULTS.TAXONOMY_DIR,
-                "taxonomy_base": DEFAULTS.TAXONOMY_BASE,
-            },
-            "evaluate": {
-                "model": DEFAULTS.DEFAULT_MODEL,
-                "taxonomy_path": DEFAULTS.TAXONOMY_DIR,
-                "taxonomy_base": DEFAULTS.TAXONOMY_BASE,
-            },
-            "train": {
-                "model_path": DEFAULTS.MODEL_REPO,
-                "data_path": DEFAULTS.DATASETS_DIR,
-                "ckpt_output_dir": DEFAULTS.CHECKPOINTS_DIR,
-                "data_output_dir": DEFAULTS.INTERNAL_DIR,
-                "max_seq_len": 4096,
-                "max_batch_len": 10000,
-                "num_epochs": 10,
-                "effective_batch_size": 3840,
-                "save_samples": 250000,
-                "lora_quantize_dtype": None,
-                "lora_rank": 4,
-                "nproc_per_node": 1,
-                "deepspeed_cpu_offload_optimizer": False,
-                "additional_args": {},
-                "is_padding_free": False,
-            },
-            "version": CONFIG_VERSION,
-        }
-
-        for key, val in defaults.items():
-            if values.get(key) is None and val is not None:
-                values[key] = val
-
-        return values
-
-
-def finish_cfg_section(
-    defaults: dict, values: typing.Dict[str, typing.Any]
-) -> typing.Dict[str, typing.Any]:
-    for key, val in defaults.items():
-        if values.get(key) is None and val is not None:
-            values[key] = val
-
-    return values
-
 
 def get_default_config() -> Config:
     """Generates default configuration for CLI"""
-    return Config(
-        version=CONFIG_VERSION,
-        chat=_chat(
-            model=DEFAULTS.DEFAULT_MODEL,
-        ),
-        generate=_generate(
-            model=DEFAULTS.DEFAULT_MODEL,
-            taxonomy_path=DEFAULTS.TAXONOMY_DIR,
-            taxonomy_base=DEFAULTS.TAXONOMY_BASE,
-        ),
-        serve=_serve(
-            model_path=DEFAULTS.DEFAULT_MODEL,
-            llama_cpp=_serve_llama_cpp(
-                gpu_layers=-1,
-                max_ctx_size=4096,
-                llm_family="",
-            ),
-            vllm=_serve_vllm(
-                llm_family="",
-                max_startup_attempts=120,
-            ),
-        ),
-        train=_train(
-            model_path=DEFAULTS.MODEL_REPO,
-            data_path=DEFAULTS.DATASETS_DIR,
-            ckpt_output_dir=DEFAULTS.CHECKPOINTS_DIR,
-            data_output_dir=DEFAULTS.INTERNAL_DIR,
-            max_seq_len=4096,
-            max_batch_len=10000,
-            num_epochs=10,
-            effective_batch_size=3840,
-            save_samples=250000,
-            lora_quantize_dtype="nf4",
```

Comment:
@cdoern about this one, the default was `nf4` but then overridden later with None, which one do you want to keep? Thanks